pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SchemaUtils.java

```diff
@@ -95,8 +101,14 @@ public static void validate(Schema schema, List<TableConfig> tableConfigs) {
    * 6) Schema validations from {@link Schema#validate}
    */
   public static void validate(Schema schema) {
+    validate(schema, false);
+  }
+
+  public static void validate(Schema schema, boolean isIgnoreCase) {
     schema.validate();
 
+    List allColumnName = schema.getColumnNames().stream().map(column -> column.toLowerCase())
+      .collect(Collectors.toList());
```

Comment:
We need to create this List only when `isIgnoreCase` is set to `true`. To make it more efficient, we can simply check while adding columns
```suggestion
    if (isIgnoreCase) {
      Set<String> lowerCaseColumnNames = new HashSet<>;
      for (String column : schema.getColumnNames()) {
        Preconditions.checkState(lowerCaseColumnNames.add(column.toLowerCase()), ...);
      }
    }
    List allColumnName = schema.getColumnNames().stream().map(column -> column.toLowerCase())
      .collect(Collectors.toList());
```

Reply:
LGTM, I will update my pr to make it more efficient.